### DISCUSSION TITLE
Favorite menus per mode

### DISCUSSION THREAD

**Discussion by madprops** [madprops/grasshopper#147] [General]
Favorites can have a mode menu that overrides the default menu.

These mode menus are empty by default, if you give them items they override the global/normal menu.

This can be useful if you want to customize modes further.

**Comment by madprops** [comment]
Made the ability to copy addlists to other addlists.

For instance you can drag the favorites menu into the list below it, and it will copy the items after a confirm.

**Comment by N3C2L** [comment]
Thanks for this awesome favorites updates, they are working good in every mode! 
However, there is a menu bug in all 3 modes: bookmarks, history, closed: Right click an item in every mode shows not the same menus that are listed here in settings > menus. If you add some cmds there in these 3 modes menus, the cmds are not visible in the modes afterwards if you right click in the item list in these modes.

**Reply by madprops** [reply]
Favorites menus are different to the actions menu, not sure what you mean.

**Reply by N3C2L** [reply]
Sorry, I forgot to add it's not in the favorites bar. What I mean is the context menu in the filtered item list. For example if I filter something in the history mode, and then right click some of the results. That these context menus on all 3 modes are bugged. Please add some cmds to them for example, navigate to settings > menus > history menu and add some cmds, and then try to right click a result. Same problem on bookmarks menu and closed menu there.

**Reply by madprops** [reply]
Actions menu is not the item menu ... I guess this is confusing

You can't define item menus unless you use the extra menu

Item menu is automatic .. i guess i should add the ability for item menus per mode

**Reply by madprops** [reply]
This should be clearer.

![menus234](https://github.com/user-attachments/assets/a6fb50f5-bf43-451a-bda6-ae047c75f85f)

Added the item menus. And now globals only kick in when a mode has no definitions.

**Comment by N3C2L** [comment]
Ah, thanks for these changes!
You are right, this is a lot clearer now!
